gradle-plugins/model-compiler/src/main/java/io/spine/gradle/compiler/util/MessageFinder.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.gradle.compiler.util;
+
+import com.google.common.base.Predicate;
+import com.google.protobuf.DescriptorProtos.DescriptorProto;
+import com.google.protobuf.DescriptorProtos.FileDescriptorProto;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static com.google.common.collect.Lists.newLinkedList;
+
+/**
+ * Utilities for finding Protobuf {@link MessageContext message declarations}.
+ *
+ * @author Dmytro Grankin
+ */
+public class MessageFinder {
+
+    private MessageFinder() {
+        // Prevent instantiation of this utility class.
+    }
+
+    /**
+     * Obtains message contexts for the messages, that match the specified {@link Predicate}.
+     *
+     * @param fileDescriptors the file descriptors to scan
+     * @param predicate       the predicate to test a message
+     * @return the message contexts
+     */
+    public static List<MessageContext> find(Collection<FileDescriptorProto> fileDescriptors,
```

Comment:
BTW, I don't understand, why we are retuning an ordered collection in this class? Is there any reason for that? If so, please include it to the documentation.